hpcgap/lib/vecmat.gd

```diff
@@ -431,6 +431,34 @@ DeclareOperation( "ImmutableMatrix",[IsObject,IsMatrix]);
 
 #############################################################################
 ##
+#F  ImmutableVector( <field>, <vector>[, <change>] ) . convert into "best" representation
+##
+##  <#GAPDoc Label="ImmutableVector">
+##  <ManSection>
+##  <Oper Name="ImmutableVector" Arg='field, vector[, change]'/>
+##
+##  <Description>
+##  returns an immutable vector equal to <A>vector</A> which is in the optimal
+##  (concerning space and runtime) representation for vectors defined over
+##  <A>field</A>. This means that vectors obtained by several calls of
+##  <Ref Oper="ImmutableVector"/> for the same <A>field</A> are compatible
+##  for fast arithmetic without need for field conversion.
+##  <P/>
+##  The input vector <A>vector</A> might change the representation,
```

Comment:
That seems inconsistent with `Immutable`, which (at least according to the documentation!) won't change representation.

Of course in 99% of cases, users shouldn't know or care representation has changed, and there are quite a lot of functions which do it without telling users, so we could alternatively just not mention it, to avoid confusion.

I assume the long-term plan would be that we wouldn't do an in-place transformation if the object is read-only or immutable (so we don't create problems in HPC-GAP)?